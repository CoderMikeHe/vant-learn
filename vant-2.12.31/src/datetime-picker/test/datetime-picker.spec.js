import DatetimePicker from '..';
import { mount } from '../../../test';

test('confirm & cancel event', () => {
  const onConfirm = jest.fn();
  const onCancel = jest.fn();

  const wrapper = mount(DatetimePicker, {
    listeners: {
      confirm: onConfirm,
      cancel: onCancel,
    },
  });

  wrapper.find('.van-picker__confirm').trigger('click');
  expect(onConfirm).toHaveBeenCalledTimes(1);

  wrapper.find('.van-picker__cancel').trigger('click');
  expect(onCancel).toHaveBeenCalledTimes(1);
});

test('time type', () => {
  const wrapper = mount(DatetimePicker, {
    propsData: {
      type: 'time',
      minHour: 22,
      minMinute: 58,
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('getPicker method', () => {
  const wrapper = mount(DatetimePicker);
  expect(wrapper.vm.getPicker()).toBeTruthy();
});

test('should render title slot correctly', () => {
  const wrapper = mount(DatetimePicker, {
    propsData: {
      showToolbar: true,
    },
    scopedSlots: {
      title: () => 'Custom title',
    },
  });

  expect(wrapper.find('.van-picker__toolbar')).toMatchSnapshot();
});
