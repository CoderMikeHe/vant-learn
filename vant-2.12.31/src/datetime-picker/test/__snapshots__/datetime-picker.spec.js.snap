// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render title slot correctly 1`] = `<div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button>Custom title<button type="button" class="van-picker__confirm">确认</button></div>`;

exports[`time type 1`] = `
<div class="van-picker van-datetime-picker">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">22</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">23</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">58</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">59</div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;
